apps/aklite-offline/cmds.cpp

```diff
@@ -2,166 +2,139 @@
+int InstallCmd::installUpdate(const po::variables_map& vm, const boost::filesystem::path& src_dir,
                               bool force_downgrade) const {
-  int ret_code{EXIT_FAILURE};
-  try {
-    const auto install_res = offline::client::install(
-        cfg_in, {src_dir / "tuf", src_dir / "ostree_repo", src_dir / "apps"}, nullptr, force_downgrade);
-
-    switch (install_res) {
-      case offline::PostInstallAction::Ok: {
-        std::cout << "Please run `aklite-offline run` command to start the updated Apps\n";
-        ret_code = 10;
-        break;
-      }
-      case offline::PostInstallAction::NeedRebootForBootFw: {
-        ret_code = 90;
-        std::cout
-            << "Please reboot a device to confirm a boot firmware update, and then run the `install` command again\n";
-        std::cout << "If the reboot doesn't help to proceed with the update, then make sure that "
-                     "`bootupgrade_available` is set to `0`.";
-        std::cout << "Try running `fw_setenv|fiovb_setenv bootupgrade_available 0`, reboot a device, and then run "
-                     "the `install` again";
-        break;
-      }
-      case offline::PostInstallAction::NeedReboot: {
-        ret_code = 100;
-        std::cout
-            << "Please reboot a device and execute `aklite-offline run` command "
-               "to apply installation and start the updated Apps (unless no Apps to update or dockerless system)\n";
-        break;
-      }
-      case offline::PostInstallAction::AlreadyInstalled: {
-        std::cout << "The given Target has been already installed\n";
-        ret_code = EXIT_SUCCESS;
-        break;
-      }
-      case offline::PostInstallAction::DowngradeAttempt: {
-        std::cout << "Refused to downgrade\n";
-        ret_code = 102;
-        break;
-      }
-      default:
-        LOG_ERROR << "Got invalid post install action code";
-    };
-
-  } catch (const std::exception& exc) {
-    std::cerr << "Failed to install the offline update; src-dir: " << src_dir << "; err: " << exc.what() << std::endl;
+  AkliteClient client(vm);
+  const LocalUpdateSource local_update_source{.tuf_repo = (src_dir / "tuf").string(),
+                                              .ostree_repo = (src_dir / "ostree_repo").string(),
+                                              .app_store = (src_dir / "apps").string()};
+  auto ret_code{aklite::cli::Install(client, -1, "", "", force_downgrade, &local_update_source)};
+  switch (ret_code) {
+    case aklite::cli::StatusCode::InstallAppsNeedFinalization: {
+      // TBD: The former `aklite-offline` sets `10` as a an exit/status code, while
+      // the current version returns `InstallAppsNeedFinalization = 105`.
+      // Maybe it makes sense to override it with `10`, but the `10` is already used for `TufMetaPullFailure = 10`?
+      std::cout << "Please run `aklite-offline run` command to start the updated Apps\n";
```

Comment:
IMO it is ok to change the value now to make it unique globally. Perhaps @caiotpereira can give some feedback on the matter.

Reply:
ok. I'll start discussion with Caio offline.